src/components/tree-item/tree-item.tsx

```diff
@@ -496,18 +496,37 @@ export class TreeItem
     items.forEach((item) => (item.parentExpanded = expanded));
   };
 
-  private updateAncestorTree = (): void => {
-    if (this.selected && this.selectionMode === "ancestors") {
-      const ancestors: HTMLCalciteTreeItemElement[] = [];
-      let parent = this.parentTreeItem;
-      while (parent) {
-        ancestors.push(parent);
-        parent = parent.parentElement?.closest("calcite-tree-item");
-      }
-      ancestors.forEach((item) => (item.indeterminate = true));
+  /**
+   * This is meant to be called in `componentDidLoad` in order to take advantage of the hierarchical component lifecycle
+   * and help check for item selection as items are initialized
+   *
+   * @private
+   */
+  private updateAncestorTree(): void {
```

Comment:
Do we have an issue, or do we want to open one, where the `tree` is in in charge of handling selection instead of individual `tree-items`? It would be a refactor but might be beneficial for the future.

Reply:
Yeah, I was thinking the same. I'll create a refactor issue for tracking purposes. Tree is a beast for sure. 👹